### ISSUE TITLE
CNN Conv Layer output some times mismatch with expected formula, please help!

### ISSUE THREAD

**Issue by siwei0729** [tensorflow/tfjs#536]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### Describe the problem or feature request
I am doing a project about machine learning visualisation, but the conv layer output confuses me a lot.

Generally for a cnn convolution layer, the output should follow the formula:
`output[ x , y ] = Relu( np.sum( input[ x : x + 5 , y : y + 5 , : ] * W ) + Bias)`

The strange thing is when filter size is 1 everything is okay. However, when filter size is 2 or larger, the output does not match the formula any more. (The result is still converge and can get good accuracy)

When only 1 filter:
<img width="1143" alt="screen shot 2018-07-22 at 12 13 02 pm" src="https://user-images.githubusercontent.com/14250127/43042170-b403f15e-8da8-11e8-8e81-04d2a71c46fe.png">
 
When 2 filters
<img width="1142" alt="screen shot 2018-07-22 at 12 13 30 pm" src="https://user-images.githubusercontent.com/14250127/43042172-c29951dc-8da8-11e8-8607-3c78df38178a.png">

#### Code to reproduce the bug / link to feature request
I have created a simple repo to reproduce this issue.
Code is here:

```
import * as tf from '@tensorflow/tfjs';
import {MnistData} from './data';

function setupModel(filterSize){
  let model = tf.sequential();
  model.add(tf.layers.conv2d({
    inputShape: [28, 28, 1],
    kernelSize: 5,
    filters: filterSize,
    strides: 1,
    activation: 'relu',
    kernelInitializer: 'varianceScaling'
  }));

  const LEARNING_RATE = 0.15;
  const optimizer = tf.train.sgd(LEARNING_RATE);
  model.compile({
    optimizer: optimizer,
    loss: 'categoricalCrossentropy',
    metrics: ['accuracy'],
  });

  return model;
}


async function showPredictions(filterSize) {
  const batch = data.nextTestBatch(1);
  let model = setupModel(filterSize);
  tf.tidy(() => {

    const input = batch.xs.reshape([-1, 28, 28, 1]);
    const output = model.predict(input);

    const inputData = input.dataSync();
    const outputData = output.dataSync();
    console.log("filterSize",filterSize);
    // console.log("input",inputData);
    // console.log("output",outputData);


    //fixed input size 28x28, no padding, strides[1,1], kernel size 5
    //Going to show first activation with inputs and weights and ouput
    for(let y = 0; y < 24; y++){
      for(let x = 0; x < 24; x++){
        let index = y * 24 + x;

        if(outputData[index] > 0){
          console.log(`First activation at [${x},${y}]`)

          let firstFilterWight = model.layers[0].getWeights()[0].dataSync().slice(0,25);
          console.log("weights of first filter",firstFilterWight);
      
          let inputWindow = []
          for(let yKernel = 0; yKernel < 5; yKernel++ ){
            let inputIndex = y * 28 + yKernel * 28 + x ;
            inputWindow.push(...inputData.slice(inputIndex,inputIndex+5));
          }
          console.log("Input window values",inputWindow);

          let dotProduct = 0;
          for(let i in inputWindow){
            dotProduct += inputWindow[i] * firstFilterWight[i];
          }


          //bias
          let bias = model.layers[0].getWeights()[1].dataSync()[0];
          dotProduct += bias;

          console.log("dot product",dotProduct);
          console.log(`First activation at [${x},${y}]`,outputData[index])

          return;
        }


      }
    }


  });
}

let data;
async function load() {
  data = new MnistData();
  await data.load();
}

async function mnist() {
  await load();
  showPredictions(1);
  showPredictions(2);
}

mnist();
```

Here is the result:
<img width="753" alt="screen shot 2018-07-22 at 12 01 13 pm" src="https://user-images.githubusercontent.com/14250127/43042184-39ef7dd8-8da9-11e8-9dd3-414734ca8f6a.png">
As you can see when filter size is 1, dot prodcut matches output, but when filter size >=2 it does not match.

[Repo is here](https://github.com/siwei0729/tfjs_CNN_example2)
Clone the repo then run 
```
yarn
yarn watch
```
